### ISSUE TITLE
data in gatsby graphql does not match data in wordpress graphql

### ISSUE THREAD

**Issue by tsdexter** [gatsbyjs/gatsby#35597]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Data from `wordpress.site/graphql` doesn't not match the same query in `gatsby.dev/___graphql`

### Reproduction Link

I can provide staging and git access if required

### Steps to Reproduce

Make a call in Wordpress graphiql and the same call in gatsby graphiql which should return the same data

### Expected Result

Here is the results of the call in wordpress graphiql backend:

<img width="869" alt="image" src="https://user-images.githubusercontent.com/4126987/166985427-82f1dfb0-0152-4f58-aaec-66b558282220.png">

### Actual Result

Here is the result from gatsby graphiql. `seo.title` differs

<img width="917" alt="image" src="https://user-images.githubusercontent.com/4126987/166985542-4527612e-baeb-49b0-9b4b-0c1cdb8248b4.png">

### Environment

```shell
System:
    OS: macOS 12.2
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.16.1 - ~/.nvm/versions/node/v12.16.1/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v12.16.1/bin/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v12.16.1/bin/npm
  Languages:
    Python: 2.7.18 - /usr/bin/python
  Browsers:
    Chrome: 101.0.4951.54
    Firefox: 99.0.1
    Safari: 15.3
  npmPackages:
    gatsby: ^4.10.0-next.3 => 4.13.0 
    gatsby-plugin-fastify: ^0.7.4 => 0.7.5 
    gatsby-plugin-gatsby-cloud: ^4.3.0 => 4.3.0 
    gatsby-plugin-image: ^2.10.0-next.3 => 2.13.0 
    gatsby-plugin-manifest: ^4.3.0 => 4.3.0 
    gatsby-plugin-offline: ^5.3.0 => 5.3.0 
    gatsby-plugin-postcss: ^5.3.0 => 5.3.0 
    gatsby-plugin-react-helmet: ^5.3.0 => 5.3.0 
    gatsby-plugin-sharp: ^4.10.0-next.3 => 4.13.0 
    gatsby-source-filesystem: ^4.3.0 => 4.3.0 
    gatsby-source-wordpress: 6.14.0-alpha-wp-exclude-media-items.0 => 6.14.0-alpha-wp-exclude-media-items.0+76fff31f66 
    gatsby-transformer-sharp: ^4.10.0-next.2 => 4.13.0 
  npmGlobalPackages:
    gatsby-cli: 3.0.0
    gatsby: 3.0.0
```

**Comment by tsdexter** [comment]
I cleaned gatsby cache, deleted `.wordpress-cache` and rebuilt, all data re-sourced and still not matching

**Comment by adkanojia** [comment]
Could this be from server cache?

**Comment by tsdexter** [comment]
@adkanojia I don't think so, the server seems to be returning the correct data but at some point during adding the SEO data to the gatsby data layer it seems to be messing up the connection between `post` -> `seo`

**Comment by TylerBarnes** [comment]
Hi, thanks for the report! If this bug is on the WPGraphQL side (not sure yet), the query comparison you're making wouldn't show it because source-wordpress uses list queries to source data, not individual node queries.
Can you [follow this guide](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-wordpress/docs/debugging-and-troubleshooting.md#missing-data-in-gatsby) to debug the issue and let me know if you still notice a discrepancy between the two? 
In the past I've seen bugs on the WPGraphQL side where wrong data was returned but only when querying in certain contexts or when also querying for specific sibling fields.

**Comment by tsdexter** [comment]
@TylerBarnes I've tracked this down to a line in the WPGrapql SEO extension that uses `for_post` from the Yoast Plugin

`$meta = YoastSEO()->meta->for_post($post->ID);`

It appears that this `for_post` was not looking up the correct post data **only when used in a loop** so when I was querying for the single page I noticed was returning incorrect data in Wordpress graphiql it returned the correct data. When I got the actual sourcing query from `writeQueriesToDisk` and ran that in wordpress graphiql and/or GraphQL PLayground it returned the incorrect data. 

Updating to Yoast 19 seems to have fixed the issue.
